File: spec/python/test_repeat_eos_struct.py
import unittest

from repeat_eos_struct import RepeatEosStruct

class TestRepeatEosStruct(unittest.TestCase):
    def test_repeat_eos_struct(self):
        r = RepeatEosStruct.from_file("src/repeat_eos_struct.bin")

        self.assertEquals(len(r.chunks), 2)
        self.assertEquals(r.chunks[0].offset, 0)
        self.assertEquals(r.chunks[0].len, 0x42)
        self.assertEquals(r.chunks[1].offset, 0x42)
        self.assertEquals(r.chunks[1].len, 0x815)

File: spec/lua/test_repeat_eos_struct.lua
local luaunit = require("luaunit")

require("repeat_eos_struct")

TestRepeatEosStruct = {}

function TestRepeatEosStruct:test_repeat_eos_struct()
    local r = RepeatEosStruct:from_file("src/repeat_eos_struct.bin")

    luaunit.assertEquals(#r.chunks, 2)
    luaunit.assertEquals(r.chunks[1].offset, 0)
    luaunit.assertEquals(r.chunks[1].len, 0x42)
    luaunit.assertEquals(r.chunks[2].offset, 0x42)
    luaunit.assertEquals(r.chunks[2].len, 0x815)
end
